code/src/zarr_converter/zarr_converter.py

```diff
@@ -196,43 +233,82 @@ def convert(
+                    'tcp': '60s'
+                },
+                'distributed.scheduler.bandwidth': 100000000,
+                # 'distributed.scheduler.unknown-task-duration': '15m',
+                # 'distributed.scheduler.default-task-durations': '2h',
+            }
+        )
+        
+        # print(dask.config.config)
+        
+        cluster = LocalCluster()
+        # cluster.adapt(
+        #     minimum=1, maximum=4, interval='10s', target_duration='60s'
+        # )
+        client = Client(cluster)
         
         image = self.read_multichannel_image(self.input_data)
         
-        if not isinstance(image, dask.array.core.Array):
-            image = self.read_channel_image(self.input_data)
-            
+        # Reading single channel
+        if not isinstance(image, dask.array.core.Array) and not isinstance(image, list):
+            image = self.pad_array_n_d(
+                self.read_channel_image(str(self.input_data)+'/*/*/'),
+                4
+            )
+        
         scale_axis = []
-        for axis in range(len(image.shape)-len(writer_config['scale_factor'])):
+        for axis in range(len(image[0].shape)-len(writer_config['scale_factor'])):
             scale_axis.append(1)
         
         scale_axis.extend(list(writer_config['scale_factor']))
         scale_axis = tuple(scale_axis)
         
-        pyramid_data = self.compute_pyramid(
-            image, 
-            writer_config['pyramid_levels'],
-            scale_axis
-        )
-        
-        pyramid_data = [self.pad_array_5d(pyramid) for pyramid in pyramid_data]
-        
-        dask_jobs = self.writer.write_multiscale(
-            pyramid=pyramid_data,  # : types.ArrayLike,  # must be 5D TCZYX
-            image_name=image_name,  #: str,
-            physical_pixel_sizes=None,
-            channel_names=None,
-            channel_colors=None,
-            scale_factor=scale_axis,  # : float = 2.0,
-            chunks=pyramid_data[0].chunksize,#writer_config['chunks'],
-            storage_options=self.opts,
-            compute_dask=False
-        )
+        # chunks = [1, image[0].shape[-2]//2, image[0].shape[-1]//2]
+        # print(chunks, scale_axis, image)
         
-        if len(dask_jobs):
-            dask_jobs = dask.persist(*dask_jobs)
-            progress(dask_jobs)
+        with performance_report(filename="dask-report.html"):
+            for idx in range(len(image)):
+                # Rechunking image chunks
+                # image[idx] = image[idx].rechunk(tuple(chunks))
+                # print("Number of partitions: ", image[idx].npartitions)
+                
+                pyramid_data = self.compute_pyramid(
+                    image[idx], 
+                    writer_config['pyramid_levels'],
+                    scale_axis
+                )
+
+                pyramid_data = [self.pad_array_n_d(pyramid) for pyramid in pyramid_data]
+
+                print(pyramid_data)

```

Comment:
stray print, or if not, prefer to use `logging`

Reply:
Yeah, I have a lot of stray prints in this file since I was testing the chunk size for the OMEZarr. I'll remove them, thanks